integration_tests/src/main/python/array_test.py

```diff
@@ -319,12 +319,14 @@ def test_array_slice_with_zero_start(data_gen, zero_start, valid_length):
     assert_gpu_and_cpu_are_equal_collect(
         lambda spark: three_col_df(spark, array_all_null_gen, zero_start_gen, valid_length_gen, length=5).selectExpr(
             f"slice(a, {zero_start}, {valid_length})"))
+    error = "The value of parameter(s) `start` in `slice` is invalid: Expects a positive or a negative value for `start`, but got" if is_spark_400_or_later \
+            else "Unexpected value for start in function slice: SQL array indices start at 1."
```

Comment:
#12339 is filed against Databricks 14.3, which this PR declares as being fixed. This only handles Upstream 4.0.0

Reply:
Fixed!